### PR TITLE
refactor: fix missing effect dependencies

### LINKED ISSUE #5874: Next.js apps: missing dependencies on useEffect
## Package
_Choose from the list:_
- [x] app-content-pages
- [x] app-project
- [x] app-root
- [ ] lib-classifier
- [ ] lib-panoptes-js
- [ ] lib-react-components
- [ ] unknown

## Describe the bug
ESLint warns when building any of the Next.js apps:
``` sh
./pages/_app.js
53:6  Warning: React Hook useEffect has missing dependencies: 'isBrowser' and 'localStorage'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps
```
## To Reproduce
`next build` any of the Next.js apps.

## Expected behavior
`useEffect` hooks **must** declare all reactive dependencies.

### PR DESCRIPTION
**Pull request by eatyourgreens**
Declare `isBrowser` and `localStorage` in the module scope, so that they aren't declared on each app render and are safe to use inside `useEffect`. Add an `initialTheme` variable to store the initial theme state when a page loads. Move the dark/light theme logic into a `usePreferredTheme` hook, which runs when the page loads.

With these changes, the initial `useEffect` hook simplifies down to:
- store the initial theme in `localStorage`.
- set the initial `themeMode` state in the browser.

_Please request review from `@zooniverse/frontend` team or an individual member of that team._ 

## Package
- app-content-pages
- app-project
- app-root

## Linked Issue and/or Talk Post
- closes #5874.

## How to Review
Build and run any of the Next.js apps.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Bug Fix
- [ ] The PR creator has listed user actions to use when testing if bug is fixed
- [ ] The bug is fixed
- [ ] Unit tests are added or updated

### CODE REVIEW

**Review by goplayoutside3 — COMMENTED** [review]
I tried running these changes in a private browser window where my theme preference is dark. It looks like a key is successfully set in localStorage, but the page does not actually refresh with dark styling for app-root or app-content-pages. Were you able to detect dark mode preference in all three apps? Such as:
- https://local.zooniverse.org:3000/about/about (app-content-pages)
- https://local.zooniverse.org:3000 (app-root)
- https://localhost.zooniverse.org:3000/projects/brooke/i-fancy-cats (app-project)
     - I also couldn't get the Classifier to load on this project run locally on this branch

**Comment by eatyourgreens** [comment]
I tested this with the PH-TESS Classify page, in dark and light mode. I'll take another look.

**Comment by eatyourgreens** [comment]
I'm seeing the same thing. `themeMode` is set to 'dark' for the About pages, but the page is light mode in the browser.

Screenshot of the About page in light mode, even though themeMode is shown as 'dark' in the browser console.

**Comment by eatyourgreens** [comment]
I added `setThemeMode` back into the `useEffect` hook, which seemed to fix the theme in the browser. For some reason, `Grommet` isn't picking up the theme without that extra `setState`.

**Comment by eatyourgreens** [comment]
>I also couldn't get the Classifier to load on this project run locally on this branch.

Something is going wrong with the Sentry client in the classifier. It's spamming Sentry with POST requests.
The PH-TESS Classify page, with the browser console open to show that the page is posting constantly to sentry.io.

**Comment by eatyourgreens** [comment]
The classifier bug is odd. I can only reproduce it by loading the classifier in a new tab, with session storage empty on first load.

https://github.com/zooniverse/front-end-monorepo/assets/59547/4a927b43-a819-4d62-9877-ee998164acca

**Comment by eatyourgreens** [comment]
There's a Sentry build error in the classifier, but I also see this error on the main branch so I don't think it's causing the problem here. The installed `@sentry` packages are on different versions.

``` sh
export 'replayIntegration' (reexported as 'replayIntegration') was not found in '@sentry/replay' (possible exports: Replay)
```
There are conflicting versions of Sentry installed: 7.91 in the classifier but 7.98 in the Next.js app.
``` sh
(base) asosxlap127:app-project odonnell$ npm explain @sentry/browser
@sentry/browser@7.91.0
node_modules/@sentry/browser
  @sentry/browser@"~7.91.0" from @zooniverse/classifier@0.0.1
  packages/lib-classifier
    @zooniverse/classifier@0.0.1
    node_modules/@zooniverse/classifier
      @zooniverse/classifier@"^0.0.1" from @zooniverse/fe-project@0.0.1
      packages/app-project
        @zooniverse/fe-project@0.0.1
        node_modules/@zooniverse/fe-project
          workspace packages/app-project from the root project
      workspace packages/lib-classifier from the root project

@sentry/browser@7.98.0
node_modules/@sentry/react/node_modules/@sentry/browser
  @sentry/browser@"7.98.0" from @sentry/react@7.98.0
  node_modules/@sentry/react
    @sentry/react@"7.98.0" from @sentry/nextjs@7.98.0
    node_modules/@sentry/nextjs
      @sentry/nextjs@"~7.98.0" from @zooniverse/fe-project@0.0.1
      packages/app-project
        @zooniverse/fe-project@0.0.1
        node_modules/@zooniverse/fe-project
          workspace packages/app-project from the root project
```

**Comment by eatyourgreens** [comment]
@goplayoutside3 Upgrading `@sentry/browser` fixes the classifier for me. All the installed `@sentry` packages must be on the same version (similar to `@babel` and `@storybook` packages.)

File: packages/app-project/pages/_app.js [hunk 1]
```diff
@@ -19,8 +19,21 @@ const GlobalStyle = createGlobalStyle`
     margin: 0;
   }
 `
-
-enableStaticRendering(typeof window === 'undefined')
+const isBrowser = typeof window !== 'undefined'
+const localStorage = isBrowser ? window.localStorage : null
+enableStaticRendering(isBrowser)
```
Comment (eatyourgreens) [inline packages/app-project/pages/_app.js:24]: I think this is why the classifier is broken. The store should be static on the server, dynamic in the browser. 

```suggestion
enableStaticRendering(!isBrowser)
```

**Comment by eatyourgreens** [comment]
D’oh, I [enabled static rendering](https://dev.to/ivandotv/mobx-server-side-rendering-with-next-js-4m18) in the browser by mistake (flipped a boolean from true to false.)

**Review by goplayoutside3 — APPROVED** [review]
Looks good!

**Pull request merged**